### PR TITLE
feat: Item Tax Templates for Germany

### PR DESCRIPTION
**Pull request by barredterra**
- Item Tax Templates for Germany

    Before:
    - One **Sales Taxes and Charges Templates** for each VAT rate: "VAT 19%", "VAT 7%", etc..
    - There were no **Item Tax Templates** and it would be useless to create them in this structure.
    - You could only select one template, and therefore tax rate, per transaction. 
    - **Tax Category** could only be assigned to only one of the templates, despite all belonging to the same category.

    After:
    - One generic **Sales Taxes and Charges Template** "VAT" with multiple **Accounts** at a default rate of 0.00.
    - **Item Tax Templates** for each tax rate are used to apply the correct rate. This way, Items with different tax rates can be sold in a single transaction.

- Validate that the provided **Company** exists before running taxes setup
- Create **Tax Category** during taxes setup

    If `tax_category` is specified in `country_wise_tax.json`, make sure the Tax Category exists before creating a Tax Template.

- [x] port to `develop`.

> no-docs

### CODE REVIEW

**Comment by deepeshgarg007** [comment]
@barredterra Have done something similar for India in this PR https://github.com/frappe/erpnext/pull/25415

I am creating Tax Categories separately as there are a few custom fields for India and then assigning them to templates.
Any thoughts on this?

**Comment by barredterra** [comment]
@deepeshgarg007 I refactored the Tax Category setup similar to your PR: f259bf4. I think this should also work for your use case.

File: erpnext/setup/setup_wizard/data/country_wise_tax.json [hunk 1]
```diff
@@ -481,63 +481,66 @@
...
 									"tax_rate": 7.00
-								}
+								},
+								"rate": 0.00
 							}
 						]
 					},
 					{
 						"title": "Innergemeinschaftlicher Erwerb 19% Umsatzsteuer und 19% Vorsteuer",
+						"tax_category": "Innergemeinschaftlicher Erwerb 19%",
```
Comment (barredterra) [inline erpnext/setup/setup_wizard/data/country_wise_tax.json]:
```suggestion
```

**Comment by barredterra** [comment]
@deepeshgarg007 could you review this PR first, then rebase yours?

**Comment by deepeshgarg007** [comment]
@barredterra this looks fine, hence merging this. Will rebase mine

**Review by deepeshgarg007 — APPROVED** [review]

**Pull request merged**